numpy/core/src/multiarray/nditer_pywrap.c

```diff
@@ -2331,6 +2382,43 @@ npyiter_ass_subscript(NewNpyArrayIterObject *self, PyObject *op,
     return -1;
 }
 
+static PyObject *
+npyiter_enter(NewNpyArrayIterObject *self)
+{
+    if (self->iter == NULL) {
+        PyErr_SetString(PyExc_ValueError, "operation on  non-initialized iterator");
+        return NULL;
+    }
+    if (self->managed != CONTEXT_NOTENTERED) {
+        PyErr_SetString(PyExc_ValueError, "cannot reenter managed nditer context");
+        return NULL;
+    }
+    self->managed = CONTEXT_INSIDE;
+    Py_INCREF(self);
+    return (PyObject *)self;
+}
+
+static PyObject *
+npyiter_exit(NewNpyArrayIterObject *self, PyObject *args)
+{
+    int iop, nop;
+    PyArrayObject **object;
+    NpyIter *iter = self->iter;
+    if (iter == NULL)
+        Py_RETURN_NONE;
+    nop = NpyIter_GetNOp(iter);
+    object = NpyIter_GetOperandArray(iter);
+    for(iop = 0; iop < nop; ++iop, ++object) {
+        if (PyArray_ResolveWritebackIfCopy(*object) < 0) {
```

Comment:
Should this call `Resolve` even if an error occurs?

Reply:
correct, I did not properly handle the error condition. TODO

Reply:
fixed in  0c3340936f3